aws-rds-dbclusterendpoint/src/test/java/software/amazon/rds/dbclusterendpoint/AbstractHandlerTest.java

```diff
@@ -0,0 +1,222 @@
+                        software.amazon.awssdk.services.rds.model.Tag.builder().key("resource-tag-2").value("resource-tag-value2").build(),
+                        software.amazon.awssdk.services.rds.model.Tag.builder().key("resource-tag-3").value("resource-tag-value3").build()
+                )).build();
+    }
+    protected static final ResourceModel RESOURCE_MODEL;
+    protected static final ResourceModel RESOURCE_MODEL_WITH_TAGS;
+
+    protected static final DBClusterEndpoint DB_CLUSTER_ENDPOINT_AVAILABLE;
+    protected static final DBClusterEndpoint DB_CLUSTER_ENDPOINT_CREATING;
+    protected static final DBClusterEndpoint DB_CLUSTER_ENDPOINT_DELETING;
+    protected static final String LOGICAL_IDENTIFIER = "DBClusterEndpointLogicalId";
+
+    protected abstract BaseHandlerStd getHandler();
+
+    protected abstract AmazonWebServicesClientProxy getProxy();
+
+    protected abstract ProxyClient<RdsClient> getRdsProxy();
+
+    static {
+        RESOURCE_MODEL = ResourceModel.builder()
+                .dBClusterEndpointIdentifier("dbClusterEndpointIdentifier")
+                .dBClusterIdentifier("clusterIdentifier")
+                .endpointType("ANY")
+                .build();
+
+        RESOURCE_MODEL_WITH_TAGS = ResourceModel.builder()
+                .dBClusterIdentifier("dbClusterEndpointIdentifier")
+                .dBClusterIdentifier("clusterIdentifier")
+                .endpointType("ANY")
+                .tags(TAG_LIST_ALTER)
+                .build();
+
+        DB_CLUSTER_ENDPOINT_AVAILABLE = DBClusterEndpoint.builder()
+                .dbClusterEndpointIdentifier("dbClusterEndpointIdentifier")
+                .dbClusterIdentifier("clusterIdentifier")
+                .endpointType("ANY")
+                .status("available")
+                .build();
+
+        DB_CLUSTER_ENDPOINT_CREATING = DBClusterEndpoint.builder()
+                .dbClusterEndpointIdentifier("dbClusterEndpointIdentifier")
+                .dbClusterIdentifier("clusterIdentifier")
+                .endpointType("ANY")
+                .status("creating")
+                .build();
+
+        DB_CLUSTER_ENDPOINT_DELETING = DBClusterEndpoint.builder()
+                .dbClusterEndpointIdentifier("dbClusterEndpointIdentifier")
+                .dbClusterIdentifier("clusterIdentifier")
+                .endpointType("ANY")
+                .status("deleting")
+                .build();
+    }
+
+    static ResourceModel.ResourceModelBuilder RESOURCE_MODEL_BUILDER() {
+        return RESOURCE_MODEL_BUILDER_WITHOUT_TAGS()
+                .tags(TAG_LIST);
+    }
+
+    static ResourceModel.ResourceModelBuilder RESOURCE_MODEL_BUILDER_WITHOUT_TAGS() {
+        return ResourceModel.builder()
+                .dBClusterEndpointIdentifier(DB_CLUSTER_ENDPOINT_IDENTIFIER)
+                .endpointType(DB_CLUSTER_ENDPOINT_TYPE)
+                .dBClusterIdentifier(DB_CLUSTER_IDENTIFIER);
+    }
```

Comment:
I would inverse these mocks. Semantically, i would expect the bare-bone builder to return a minimal valid attribute set. The one with tags could be called `RESOURCE_MODEL_WITH_TAGS`. Another bit is: a model and its builder are inter-convertible.